### PR TITLE
Python dukpy lib

### PR DESCRIPTION
**Pull request by luka177**
So it compiles now

### CODE REVIEW

File: dev-python/dukpy/dukpy-0.3.recipe [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+DESCRIPTION="dukpy is a JavaScript runtime environment for Python (tested with python 2.7 and >= 3.4) using the duktape embeddable JavaScript engine. With dukpy, you can run JavaScript in Python."
+HOMEPAGE="https://github.com/kovidgoyal/dukpyt"
+COPYRIGHT="2007-2019 Ian Bicking and contributors"
+LICENSE="BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="https://github.com/kovidgoyal/dukpy/archive/v0.3.tar.gz"
+CHECKSUM_SHA256="cc78c39ebba51f381c623b164cfb7dcf3caddf515fe7094bc53b7eca5d4e435e"
+
+
+ARCHITECTURES="any"
```
Comment (Begasus) [inline dev-python/dukpy/dukpy-0.3.recipe]: As said on IRC, architecture can't be any here, but seeing you focus on x86_64 try to change this to:
```diff
-ARCHITECTURES="any"
+ARCHITECTURES="!x86_gcc2 x86_64"
```

Comment (Begasus) [reply dev-python/dukpy/dukpy-0.3.recipe]: Would be nice if you could check this on gcc2h (32bit) and fix it)

File: dev-python/dukpy/dukpy-0.3.recipe [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+PROVIDES="
+	$portName = $portVersion
+	"
+REQUIRES="
+	haiku
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	gcc
```
Comment (Begasus) [inline dev-python/dukpy/dukpy-0.3.recipe]: gcc should go into BUILD_REQUIRES:
```diff
-	gcc
```

File: dev-python/dukpy/dukpy-0.3.recipe [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	"
+
```
Comment (Begasus) [inline dev-python/dukpy/dukpy-0.3.recipe]: Add gcc cmd here:
```diff
+BUILD_PREREQUIRES="
+	cmd:gcc
+	"
```

File: dev-python/dukpy/dukpy-0.3.recipe (hunk 1 above)
Comment (luka177) [reply dev-python/dukpy/dukpy-0.3.recipe]: ok workes on 64

File: dev-python/dukpy/dukpy-0.3.recipe (hunk 2 above)
Comment (luka177) [reply dev-python/dukpy/dukpy-0.3.recipe]: fixed

File: dev-python/dukpy/dukpy-0.3.recipe (hunk 3 above)
Comment (luka177) [reply dev-python/dukpy/dukpy-0.3.recipe]: done

**Comment by luka177** [comment]
So tested it, it worked ok

File: dev-python/dukpy/dukpy-0.3.recipe [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+PROVIDES="
+	$portName = $portVersion
+	"
+REQUIRES="
+	haiku
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	cmd:gcc
```
Comment (Begasus) [inline dev-python/dukpy/dukpy-0.3.recipe]: cmd: gcc shouldn't be needed here, it should be moved to BUILD_PREREQUIRES

Comment (luka177) [reply dev-python/dukpy/dukpy-0.3.recipe]: like now?

Comment (luka177) [reply dev-python/dukpy/dukpy-0.3.recipe]: or like this?

Comment (luka177) [reply dev-python/dukpy/dukpy-0.3.recipe]: nope

File: dev-python/dukpy/dukpy-0.3.recipe [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+CHECKSUM_SHA256="cc78c39ebba51f381c623b164cfb7dcf3caddf515fe7094bc53b7eca5d4e435e"
+
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+
+PROVIDES="
+	$portName = $portVersion
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc	"
```
Comment (Begasus) [inline dev-python/dukpy/dukpy-0.3.recipe]: This should be moved after BUILD_REQUIRES, check https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines

Comment (luka177) [reply dev-python/dukpy/dukpy-0.3.recipe]: ouch

Comment (luka177) [reply dev-python/dukpy/dukpy-0.3.recipe]: like now?

File: dev-python/dukpy/dukpy-0.3.recipe [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+REQUIRES="
+	haiku
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+BUILD_PREREQUIRES="$BUILD_PREREQUIRES
+	cmd:gcc"
+PYTHON_PACKAGES=(python36 python3)
```
Comment (Begasus) [inline dev-python/dukpy/dukpy-0.3.recipe]: Move the " to the next line after a tab, insert a new line between this block and the next one:
```diff
-	cmd:gcc"
-PYTHON_PACKAGES=(python36 python3)
+	cmd:gcc
+	"
+
PYTHON_PACKAGES=(python36 python3)

Comment (luka177) [reply dev-python/dukpy/dukpy-0.3.recipe]: We need cmg:gcc here?

Comment (Begasus) [reply dev-python/dukpy/dukpy-0.3.recipe]: Left a missing remark, think it should be clear now

Comment (luka177) [reply dev-python/dukpy/dukpy-0.3.recipe]: like this?

File: dev-python/dukpy/dukpy-0.3.recipe [hunk 7]
```diff
@@ -0,0 +1,71 @@
+SUMMARY="JavaScript in Python"
+DESCRIPTION="dukpy is a JavaScript runtime environment for Python (tested with python 2.7 and >= 3.4) using the duktape embeddable JavaScript engine. With dukpy, you can run JavaScript in Python."
```
Comment (Begasus) [inline dev-python/dukpy/dukpy-0.3.recipe]: Keep description in the limit of 80 characters, check the guidelines

Comment (luka177) [reply dev-python/dukpy/dukpy-0.3.recipe]: ok fixed

**Comment by korli** [comment]
"Trailing whitespace in 'dev-python/dukpy/dukpy-0.3.recipe'."

**Comment by luka177** [comment]
> "Trailing whitespace in 'dev-python/dukpy/dukpy-0.3.recipe'."

@koril yep found at line 18 now it is ok

File: dev-python/dukpy/dukpy-0.3.recipe [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+	\"; \
+REQUIRES_$pythonPackage=\"\
+	haiku\n\
+	cmd:python$pythonVersion\
+	\""
+BUILD_REQUIRES="$BUILD_REQUIRES
+	setuptools_$pythonPackage"
+BUILD_PREREQUIRES="$BUILD_PREREQUIRES
+	cmd:python$pythonVersion"
+
```
Comment (Begasus) [inline dev-python/dukpy/dukpy-0.3.recipe]: No new line here:
```diff
-
```

File: dev-python/dukpy/dukpy-0.3.recipe [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+done
+
+PROVIDES_python36="$PROVIDES_python36
+	cmd:dukpy
+	"
+PROVIDES_python3="$PROVIDES_python3
+	cmd:dukpy
+	"
```
Comment (Begasus) [inline dev-python/dukpy/dukpy-0.3.recipe]: These shouldn't be needed:
```diff
-PROVIDES_python36="$PROVIDES_python36
-	cmd:dukpy
-	"
-PROVIDES_python3="$PROVIDES_python3
-	cmd:dukpy
-	"
```

File: dev-python/dukpy/dukpy-0.3.recipe [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+REQUIRES="
+	haiku
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+BUILD_PREREQUIRES="$BUILD_PREREQUIRES
+	cmd:gcc
+	"+SOURCE_URI="https://github.com/kovidgoyal/dukpy/archive/v$portVersion.tar.gz
```
Comment (Begasus) [inline dev-python/dukpy/dukpy-0.3.recipe]: SOURCE_URI shouldn't be here, it should stay where it was (just after line7)

File: dev-python/dukpy/dukpy-0.3.recipe [hunk 11]
```diff
@@ -0,0 +1,69 @@
+SUMMARY="JavaScript in Python"
+DESCRIPTION="dukpy is a JavaScript runtime environment for Python."
+HOMEPAGE="https://github.com/kovidgoyal/dukpyt"
+COPYRIGHT="Duktape authors"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://github.com/kovidgoyal/dukpy/archive/v0.3.tar.gz"
```
[the remark this answers is not in the document]
Comment (Begasus) [reply dev-python/dukpy/dukpy-0.3.recipe]: My bad here, forgot to include the closing " for SOURCE_URI

File: dev-python/dukpy/dukpy-0.3.recipe [hunk 12]
```diff
@@ -0,0 +1,63 @@
+SUMMARY="JavaScript in Python"
+DESCRIPTION="dukpy is a JavaScript runtime environment for Python."
+HOMEPAGE="https://github.com/kovidgoyal/dukpy"
+COPYRIGHT="Duktape authors"
```
Comment (Begasus) [inline dev-python/dukpy/dukpy-0.3.recipe]: Add year(s):
```diff
-COPYRIGHT="Duktape authors"
+COPYRIGHT="2013-2016 by Duktape authors"
```

Comment (Begasus) [reply dev-python/dukpy/dukpy-0.3.recipe]: https://github.com/kovidgoyal/dukpy/blob/016495ed9b075e4e9452da1f2b1edcc4e82b90a9/LICENSE.txt#L7

File: dev-python/dukpy/dukpy-0.3.recipe [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+	haiku
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+BUILD_PREREQUIRES="$BUILD_PREREQUIRES
+	cmd:gcc
+	"
+PYTHON_PACKAGES=(python36 python3)
```
Comment (Begasus) [inline dev-python/dukpy/dukpy-0.3.recipe]: Add New line in between:
```diff

	"
+
PYTHON_PACKAGES=(python36 python3)
```

File: dev-python/dukpy/dukpy-0.3.recipe [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+REQUIRES_$pythonPackage=\"\
+	haiku\n\
+	cmd:python$pythonVersion\
+	\""
+BUILD_REQUIRES="$BUILD_REQUIRES
+	setuptools_$pythonPackage"
+BUILD_PREREQUIRES="$BUILD_PREREQUIRES
+	cmd:python$pythonVersion"
+
+done
```
Comment (Begasus) [inline dev-python/dukpy/dukpy-0.3.recipe]: Line 42 should be removed (done is part of this function):
```
	cmd:python$pythonVersion"
done
```

**Comment by extrowerk** [comment]
Looks ok to me, import test worked on both Python 3.6 and 3.7.

**Comment by Begasus** [comment]
LGTM, only could use support for secondary architecture so we could have a package available on gcc2h too

**Comment by Begasus** [comment]
@luka177 could you take care of adding secondary architecture support?

**Pull request merged**